### DISCUSSION TITLE
String interpolation deprecation

### DISCUSSION THREAD

**Discussion by kefyru** [dotnet/roslyn#61008] [Ideas]
```cs
var v = ValueTask.FromResult(777);
var s = $"Value: {v}"; // no warnings
```
In this case it is obviously that should use `$"Value: {await v}"` but there are situations where it is not obvious...
I think, every type that has GetAwaiter() method should cause warning with string interpolation. 

Case:
```cs

var managerName = 
   invoice             // Value
      .Customer    // Value
      .Contact       // Awaiter
      .Name;         // Awaiter

Console.WriteLine($"Hello {managerName}");   // should cause warning "Use await ..."
Console.WriteLine($"Hello {await managerName}"); // ok
```

**Comment by CyrusNajmabadi** [comment]
It's not at all obvious that this is desirable. Task types may absolutely have a reasonable ToString (for example saying what the status of the task is). 

That said, if you want a warning here, it would be very easy to build one using a roslyn analyzer.
